src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/RichTextEditorTextArea.kt

```diff
@@ -0,0 +1,59 @@
+package com.atlassian.performance.tools.jiraactions.api.page
+
+import com.atlassian.performance.tools.jiraactions.Patience
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.openqa.selenium.By
+import org.openqa.selenium.JavascriptExecutor
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedConditions
+import org.openqa.selenium.support.ui.ExpectedConditions.*
+import java.time.Duration
+
+/**
+ * Helper class for Rich Text Editor enabled Textarea
+ *
+ */
+class RichTextEditorTextArea(
+    private val driver: WebDriver
+) {
+
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    private fun clearAndTypeIfPresent(id :String, text: String) {
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            driver.wait(timeout = Duration.ofSeconds(2), condition = elementToBeClickable(webElement))
+            webElement.clear()
+            webElement.sendKeys(text)
+        }
+    }
+
+    fun clearAndTypeIfPresentRte(id : String, text: String) {
+        //check richeditor-cover exists in class attribute
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            val attrClass = webElement.getAttribute("class")
+            if(attrClass != null && attrClass.contains(other = "richeditor-cover", ignoreCase = true)){
+                logger.debug("richeditor detected, will try Rich Text Editer...")
```

Comment:
This debug logging basically marks the entry into the `if`. Let's say there's a bug somewhere within the `if` and an exception is thrown. Its stacktrace will already prove that it failed within the `if`. Seems unnecessary and spammy, because it (or its `else` counterpart` will be logged for every action using Jira text field.

Reply:
yep, that's correct, was mainly for debug purpose during dev

Reply:
I recommend using the IntelliJ debugger. It a lot of power and flexibility in comparison to temporary logging. And it doesn't risk merging leftovers :)